File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/Acronym.java
package exercism;

 public class Acronym {

     private final String acronym;

     Acronym(String phrase) {
         var acronymBuilder = new StringBuilder();

         for (String word : phrase.split("[\\s_-]")) {
             if (!word.isEmpty()) {
                 acronymBuilder.append(word.charAt(0));
             }
         }

         acronym = acronymBuilder.toString().toUpperCase();
     }

     String get() {
         return acronym;
     }

 }

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/Acronym.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp;

using System;
using System.Linq;
using System.Text.RegularExpressions;
public static class Acronym
{
    public static string Abbreviate(string phrase) => string.Concat(Regex.Split(phrase, "[^A-Za-z']+").Select(s => char.ToUpper(s[0])));
}